internal/xsql/collection.go

```diff
@@ -37,6 +37,7 @@
 // Collection A collection of rows as a table. It is used for window, join, group by, etc.
 type Collection interface {
 	SortingData
+	RangeProjectSet(f func(i int, r CloneAbleRow) ([]CloneAbleRow, error)) ([]Collection, error)
```

Comment:
This seems like only projectset needs, I don't think we should put it here. This interface should only provide some basic/atomic functions.